servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -214,7 +215,7 @@ private DnsDiscoveryObserver newDiscoveryObserver(final String address) {
             return null;
         }
         try {
-            return observer.onNewDiscovery(address);
+            return observer.onNewDiscovery(id, address);
```

Comment:
Btw, while you are here, how do you think should we do in the constructor something like this?
```
this.id = id + '@' + toHexString(identityHashCode(this));
```
Then we can simplify `toString()` to be:
```java
    @Override
    public String toString() {
        return id;
    }
```
There is a risk of having multiple instances built be the same builder. All of them will have the same id. `identityHashCode` will always help the observer to see unique IDs.